altair_widgets/widget.py

```diff
@@ -232,8 +232,18 @@ def _get_types():
 
 
 def _get_encodings():
-    return ['x', 'y', 'color', 'text', 'row', 'column',
-            'opacity', 'shape', 'size']
+    # All the subclasses of altair.FieldChannelMixin, lowercase
+    encodings = []
+    for name in dir(altair):
+        value = getattr(altair, name)
+        if (type(value) is type and
+            issubclass(value, altair.FieldChannelMixin) and
+            name != 'FieldChannelMixin'):
+            encodings.append(name.lower())
+    # reorder to have the most useful encodings at the top
+    top = ['x', 'y', 'color']
+    list(map(encodings.remove, top))
```

Comment:
It'd nice to have this be a list comp; `[x for x in encodings if x not in top]`.